packages/tools/src/tools/annotation/RectangleROITool.ts

```diff
@@ -631,6 +631,11 @@ export default class RectangleROITool extends AnnotationTool {
 
     for (let i = 0; i < annotations.length; i++) {
       const annotation = annotations[i] as RectangleROIAnnotation;
+
+      if (isAnnotationHidden(annotation)) {
```

Comment:
hmmm, I think this skipping will make a problem for the cachedStats calculations and they become non-synced with the other targets. We should only skip the rendering portion of the `renderAnnotation` logic

Reply:
hm, but once the annotation is hidden it means anytime method enters here the annotation would have exactly the same value (cachedStats) as prior being hidden. The only moment I see it being out of sync is if when user is moving handle he press a short key, then it turns to hidden prior to updating the cachedStats: is that possible?

Reply:
I can follow your input, Im just checking if we are doing unnecessary maths

Reply:
For any usecase that changes the metadata for the images (e.g., SUV re-calculations based on new tags) we need to update the stats, It is safe to update stats

Reply:
ok